cmd/uplink/cmd/setup_test.go

```diff
@@ -115,3 +121,58 @@ func TestDefaultPortAppliedToSatelliteAddrWithPortColonButNoPort(t *testing.T) {
 	assert.Equal(t, "ahost:7777", setupCmd.Flags().Lookup("satellite-addr").Value.String(),
 		"satellite-addr should contain default port when no port specified")
 }
+
+func TestSaveEncriptionKey(t *testing.T) {
+	expectedKey := make([]byte, rand.Intn(20)+1)
+	_, err := rand.Read(expectedKey)
+	require.NoError(t, err)
+
+	t.Run("ok", func(t *testing.T) {
+		ctx := testcontext.New(t)
+		defer ctx.Cleanup()
+
+		filename := ctx.File("storj-test-cmd-uplink", ".enc.key")
+		err = cmd.SaveEncryptionKey(expectedKey, filename)
+		require.NoError(t, err)
+
+		// Read the key from the file to compare that it matches with the saved one.
+		file, err := os.Open(filename)
+		require.NoError(t, err)
+		defer func() { require.NoError(t, file.Close()) }()
+
+		key := make([]byte, len(expectedKey))
+		_, err = file.Read(key)
+		require.NoError(t, err)
+		assert.Equal(t, expectedKey, key)
+
+		// Check that the key file has a read only file permissions for the user
+		fileInfo, err := file.Stat()
+		require.NoError(t, err)
+		assert.Equal(t, os.FileMode(0400), fileInfo.Mode().Perm())
+	})
+
+	t.Run("error: unexisting dir", func(t *testing.T) {
```

Comment:
`t.Run("missing dir", func(t *testing.T) {`

Reply:
It's a way to mark the `t.Run` which are `ok` (i.e. happy path) and the ones which check for errors.